sklearn/base.py

```diff
@@ -33,7 +33,7 @@
     'stateless': False,
     'multilabel': False,
     '_skip_test': False,
-    '_xfail_test': False,
+    '_xfail_checks': {},
```

Comment:
I'm weary of having mutable object in this default dict. You never know what downstream object can do with it. I think it should not be mutable, hence the False. We also say somewhere in the doc that all estimator tags with a few exceptions default to False. 

Unless this is solving some issue?

Reply:
it makes the code cleaner and easier to understand because we can just do `    if check_name not in xfail_checks:` instead of checking if it's a bool first.

I share the concern about not using mutable globals, but right above there is the `    'X_types': ['2darray']` tag so I thought we could just do it here as well

Alternatively I can update `_safe_tags()` to return `{}` when the tag is False?

Reply:
> I share the concern about not using mutable globals, but right above there is the 'X_types': ['2darray'] tag so I thought we could just do it here as well

Indeed, but it one puts wrong either all tests are going to be skipped or some would fail. Here it would potentially skip one test which is harder to notice.

> Alternatively I can update _safe_tags() to return {} when the tag is False?

+1 for this particular tag.